### DISCUSSION TITLE
Meshed Target

### DISCUSSION THREAD

**Discussion by phy-ak** [epochpic/epoch#559] [Q&A]
How to construct the periodically arranged square meshes with period size d = 10 and 12 micro meter, so that thickness of the boron filled in the x-direction D is about 16 micron?

**Comment by Status-Mirror** [comment]
Hey @Arun-lime,

What do you mean by "square meshes"? Are you asking for a 2D geometry where you have a grid of $(16 \times 16) \mu m^2$ squares of boron, surrounded by a $10\mu m$ vacuum gap? This could be achieved using the EPOCH [maths parser](https://epochpic.github.io/documentation/code_details/maths_parser.html), and would look something like this:

![Square_mesh](https://github.com/Warwick-Plasma/epoch/assets/38282483/0307a215-a176-4a40-93ec-05683b15e1d7)

The input deck which generated this data is present here:

```
begin:control
  nx =  100
  ny =  100
  t_end = 10 * femto
  x_min = 0 * micron
  x_max =  100 * micron
  y_min = 0 * micron
  y_max =  100 * micron
  stdout_frequency = 50
end:control

begin:boundaries
  bc_x_min = open
  bc_x_max = open
  bc_y_min = open
  bc_y_max = open
end:boundaries

begin:constant
  # Set grid size
  x_thickness = 16.0 * micron
  x_gap = 10.0 * micron
  y_thickness = 16.0 * micron 
  y_gap = 10.0 * micron
end:constant

begin:constant
  # Logical test for cells within grid
  x_tot = x_thickness + x_gap
  y_tot = y_thickness + y_gap
  x_remainder = (x / x_tot) - floor(x / x_tot)
  y_remainder = (y / y_tot) - floor(y / y_tot)
  boron_frac_x = x_thickness / x_tot
  boron_frac_y = y_thickness / y_tot
end:constant

begin:species
  name = Boron
  charge = 0
  mass   = 19707.2
  number_density = if((x_remainder lt boron_frac_x) and (y_remainder lt boron_frac_y), 1.0e10, 0)
  npart_per_cell = 10
end:species

begin:output
  dt_snapshot = 10 * femto
  number_density = always
end:output
```

**Comment by DanRRRR** [comment]
Wow, that was hell impressive method ! 
Such methods have to be collected as do the books for famous chess openings :)

**Comment by phy-ak** [comment]
Dear Stuart, 
    The schematic diagram is attached below.
![Screenshot from 2023-10-12 10-57-18](https://github.com/Warwick-Plasma/epoch/assets/138562383/8a4632ee-70be-446b-8186-45d23d128903)
If period size d =10 and 12 microns, isnt it y_gap should 12 micron?
